### PR TITLE
Sg/python etl

### PR DESCRIPTION
**Pull request by saiemgilani**
This is the updated functionality with the current year's schedule as of now. While all games are listed at midnight, it's not failing, at least.
image

- ~The majority of it is adding the data to the repository~ 
- adding data/* to .dockerignore file.
- make light fixes to cfb_pbp initial data pull function so that it would fail less often when a piece is missing for a given game. 
- Add download/retry function to prevent failures from 500 errors
- update schedule.json file for 2021 schedule links

### CODE REVIEW

File: python/play_handler.py [hunk 1]
```diff
@@ -231,6 +233,19 @@ def __init__(self, logger = None, gameId = 0):
         self.logger = logger
         self.ran_pipeline = False
 
+    def download(self, url, num_retries=5):
```
Comment (akeaswaran) [inline python/play_handler.py:236]: Where are you using this method? it just seems to be recursively calling itself without actually being used anywhere

File: python/play_handler.py [hunk 2]
```diff
@@ -251,8 +266,9 @@ def cfb_pbp(self):
         pbp_txt['espnWP'] = np.array([])
         pbp_txt['gameInfo'] = np.array([])
         pbp_txt['season'] = np.array([])
-        pbp_txt['timeouts'] = {}
         pbp_txt = json.loads(pbp_resp.read())['gamepackageJSON']
+        
+        pbp_txt['timeouts'] = {}
         # summary endpoint for pickcenter array
         summary_url = "http://site.api.espn.com/apis/site/v2/sports/football/college-football/summary?event={}".format(self.gameId)
         summary_resp = urllib.request.urlopen(summary_url)
```
Comment (akeaswaran) [inline python/play_handler.py:274]: I assume this is where you want to use that `download()` method

File: python/play_handler.py [hunk 3]
```diff
@@ -647,6 +688,17 @@ def cfb_pbp(self):
             del pbp_txt['plays']['clock.mm']
         else:
             pbp_txt['drives']={}
+        
+        if 'scoringPlays' not in pbp_txt.keys():
+            pbp_txt['scoringPlays']=np.array([])
+        if 'winprobability' not in pbp_txt.keys():
+            pbp_txt['winprobability'] = np.array([])
```
Comment (akeaswaran) [inline python/play_handler.py:695]: I don't think you need to use `np.array()`, just use `[]`

Comment (saiemgilani) [reply python/play_handler.py:696]: in order for them to be able to be also comfortably used with dump, you like to have ```np.array([]).tolist()``` . i'm just being extra careful,

File: python/play_handler.py (hunk 1 above)
Comment (saiemgilani) [reply python/play_handler.py:236]: see [commit](https://github.com/saiemgilani/game-on-paper-app/pull/45/commits/8c4f40ddd3ef436c9150fe74cf54c5594a1d8caf)

**Review by akeaswaran — APPROVED** [review]

**Pull request merged**